sequencer/src/catchup.rs

```diff
@@ -86,16 +86,17 @@ impl StateCatchup for MockStateCatchup {
         view: ViewNumber,
         mt: &mut BlockMerkleTree,
         elem: &Commitment<Header>,
-    ) -> BlocksFrontier {
+    ) {
+        tracing::info!("catchup: fetching frontier for view {view:?}");
+        let view = view.get_u64();
         let (_, proof) = self
             .state
             .block_merkle_tree
-            .lookup(view.get_u64())
+            .lookup(view)
             .expect_ok()
             .unwrap();
-        mt.remember(view.get_u64(), elem, proof.clone())
```

Comment:
This is currently where the tests panic. The `element` returned by lookup doesn't match the one passed in to the function.